File: programmers/ggyul/level1/모의고사/Solution2.java
import java.util.*;

// 15분
class Solution {

    public int[] solution(int[] answers) {
        int[] answer = {};
        int[][] pattern = new int[][] {
                { 1, 2, 3, 4, 5 },
                { 2, 1, 2, 3, 2, 4, 2, 5 },
                { 3, 3, 1, 1, 2, 2, 4, 4, 5, 5, }
        };

        int[] scores = new int[pattern.length];
        for (int i = 0; i < answers.length; i++) {
            for (int j = 0; j < pattern.length; j++) {
                if (answers[i] == pattern[j][i % pattern[j].length])
                    scores[j]++;
            }
        }

        int max = 0;
        for (int i = 0; i < scores.length; i++) {
            if (max < scores[i])
                max = scores[i];
        }

        List<Integer> winners = new ArrayList<>();
        for (int i = 0; i < scores.length; i++) {
            if (max == scores[i])
                winners.add(i + 1);
        }
        Collections.sort(winners);

        answer = winners.stream()
                .mapToInt(i -> i)
                .toArray();

        return answer;
    }
}

File: programmers/ggyul/level1/최대공약수와_최소공배수/solution2.cpp
#include <string>
#include <vector>
using namespace std;

// a, b의 최대공약수와, b, a%b의 최대공약수는 같다.
// (a,b)->(b,a%b)->...(gcd, 0)
int GCD(int a, int b) {
    if (b == 0) return a;
    return GCD(b, a % b);
}

vector<int> solution(int n, int m) {
    vector<int> answer;
    int gcd = GCD(n, m);
    answer.push_back(gcd);
    int lcm = gcd;
    answer.push_back(n * m / gcd);
    return answer;
}